django/api/services/tests/test_calculate_rebate.py

```diff
@@ -0,0 +1,189 @@
+            ],
+        }
+        rebate_amount = calculate_rebate_amount(cra_response)
+        self.assertEqual(
+            rebate_amount, INCOME_REBATES.get(RebateType.A.value).get("rebate")
+        )
+
+    def test_household_receives_rebate_b_individual(self):
+        # qualifies for 'b' individual or 'c' household (gets b)
+        cra_response = {
+            "B5t92XeH7NnFUwxc": [
+                {"sin": "302435839", "year": "2020", "income": "85000"},
+                {"sin": "270300379", "year": "2020", "income": "70000"},
+            ],
+        }
+        rebate_amount = calculate_rebate_amount(cra_response)
+        self.assertEqual(
+            rebate_amount, INCOME_REBATES.get(RebateType.B.value).get("rebate")
+        )
+
+    def test_household_receives_rebate_b(self):
+        # qualifies for 'c' individual and 'b' household (should get b)
+        cra_response = {
+            "B5t92XeH7NnFUwxc": [
+                {"sin": "302435839", "year": "2020", "income": "95000"},
+                {"sin": "270300379", "year": "2020", "income": "50000"},
+            ],
+        }
+        rebate_amount = calculate_rebate_amount(cra_response)
+        self.assertEqual(
+            rebate_amount, INCOME_REBATES.get(RebateType.B.value).get("rebate")
+        )
+
+    def test_household_receives_rebate_b_switched_cra(self):
+        # qualifies for 'c' individual and 'b' household (should get b)
+        # just switched cra response
+        cra_response = {
+            "B5t92XeH7NnFUwxc": [
+                {"sin": "270300379", "year": "2020", "income": "50000"},
+                {"sin": "302435839", "year": "2020", "income": "95000"},
+            ],
+        }
+        rebate_amount = calculate_rebate_amount(cra_response)
+        self.assertEqual(
+            rebate_amount, INCOME_REBATES.get(RebateType.B.value).get("rebate")
+        )
+
+    def test_household_receives_rebate_a(self):
+        # too high personal income but qualifies for 'a' household
+        cra_response = {
+            "B5t92XeH7NnFUwxc": [
+                {"sin": "302435839", "year": "2020", "income": "100001"},
+                {"sin": "270300379", "year": "2020", "income": "1"},
+            ],
+        }
+        rebate_amount = calculate_rebate_amount(cra_response)
+        self.assertEqual(
+            rebate_amount, INCOME_REBATES.get(RebateType.A.value).get("rebate")
+        )
+
+    def test_household_receives_not_approved(self):
+        # too high invididual AND household income
+        cra_response = {
+            "B5t92XeH7NnFUwxc": [
+                {"sin": "302435839", "year": "2020", "income": "100002"},
+                {"sin": "270300379", "year": "2020", "income": "100100"},
+            ],
+        }
+        rebate_amount = calculate_rebate_amount(cra_response)
+        self.assertEqual(
+            rebate_amount, INCOME_REBATES.get(RebateType.D.value).get("rebate")
+        )
+
+    def test_application_no_cra(self):
```

Comment:
There's a few cases for no cra, this test might be `test_household_no_primary_income_receives_rebate_d`